### ISSUE TITLE
Can't deserialize a serialized DateTimeZone with default typing

### ISSUE THREAD

**Issue by strmer15** [FasterXML/jackson-datatype-joda#82]
Trying to serialize a DateTimeZone instance, then deserialize it back afterwards; it always fails.  Looks like it's due to the fact that there is no type mapping for CachedDateTimeZone (indeed for any of the subclasses of DateTimeZone) and no String constructor, combined with our project's use of default typing on serialized objects.

```
package com.rw;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.joda.JodaModule;
import org.joda.time.DateTimeZone;
import org.junit.Assert;
import org.junit.Test;

public class TestTimeZone {

    @Test
    public void testTimeZoneSerialization() throws Exception {
        final ObjectMapper objectMapper = new ObjectMapper();
        objectMapper.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL);

        objectMapper.registerModule(new JodaModule());

        final DateTimeZone timeZone = DateTimeZone.forID("America/New_York");
        final String json = objectMapper.writeValueAsString(timeZone);

        final DateTimeZone timeZoneRead = objectMapper.readValue(json, DateTimeZone.class);

        Assert.assertNotNull(timeZoneRead.getID());
        Assert.assertEquals("America/New_York", timeZoneRead.getID());
    }
}
```

Result:

```
com.fasterxml.jackson.databind.JsonMappingException: Can not instantiate value of type [simple type, class org.joda.time.tz.CachedDateTimeZone] from String value ('America/New_York'); no single-String constructor/factory method
 at [Source: ["org.joda.time.tz.CachedDateTimeZone","America/New_York"]; line: 1, column: 40]
```

Obviously, a simple workaround is to create a mix-in just for DateTimeZone that disables typing only for it.  The following test passes:

```
package com.rw;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.joda.JodaModule;
import org.joda.time.DateTimeZone;
import org.junit.Assert;
import org.junit.Test;

public class TestTimeZone {

    @Test
    public void testTimeZoneSerialization() throws Exception {
        final ObjectMapper objectMapper = new ObjectMapper();
        objectMapper.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL);
        objectMapper.addMixIn(DateTimeZone.class, DateTimeZoneMixin.class);

        objectMapper.registerModule(new JodaModule());

        final DateTimeZone timeZone = DateTimeZone.forID("America/New_York");
        final String json = objectMapper.writeValueAsString(timeZone);

        final DateTimeZone timeZoneRead = objectMapper.readValue(json, DateTimeZone.class);

        Assert.assertNotNull(timeZoneRead.getID());
        Assert.assertEquals("America/New_York", timeZoneRead.getID());
    }

    @JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
    public interface DateTimeZoneMixin {

    }
}
```

**Comment by cowtowncoder** [comment]
Thank you for reporting this. Is this reproducible on 2.7.2?

**Comment by strmer15** [comment]
Yep - sorry, should have mentioned which versions I tested against, here's the POM:

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">

    <modelVersion>4.0.0</modelVersion>

    <groupId>com.rw</groupId>
    <artifactId>test-jackson-joda</artifactId>
    <version>1.0</version>

    <dependencies>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-core</artifactId>
            <version>2.7.2</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-core</artifactId>
            <version>2.7.2</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-annotations</artifactId>
            <version>2.7.2</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-databind</artifactId>
            <version>2.7.2</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.datatype</groupId>
            <artifactId>jackson-datatype-joda</artifactId>
            <version>2.7.2</version>
        </dependency>
        <dependency>
            <groupId>joda-time</groupId>
            <artifactId>joda-time</artifactId>
            <version>2.9.2</version>
        </dependency>
    </dependencies>
</project>
```

**Comment by cowtowncoder** [comment]
No prob, thanks!

**Comment by cowtowncoder** [comment]
Just needed to basically force type information to use abstract type `DateTimeZone`; deserializer knows how to deal with that type. Will be included in 2.7.3, as well as 2.6.6 when they get released.
